VariantShredding.md

```diff
@@ -25,290 +25,316 @@
-* Simplified and good performance for write shredding. No need to rebuild the metadata, or re-encode IDs for `variant_value`.
-* Simplified and good performance for Variant reconstruction. No need to re-encode IDs for `variant_value`.
+| Array                            | `value`     | `typed_value `| `typed_value...value` | `typed_value...typed_value`    |
+|----------------------------------|-------------|---------------|-----------------------|--------------------------------|
+| `["comedy", "drama"]`            | null        | non-null      | [null, null]          | [`comedy`, `drama`]            |
+| `["horror", null]`               | null        | non-null      | [null, `00`]          | [`horror`, null]               |
+| `["comedy", "drama", "romance"]` | null        | non-null      | [null, null, null]    | [`comedy`, `drama`, `romance`] |
+| null                             | `00` (null) | null          |                       |                                |
 
-# Data Skipping
+#### Objects
 
-Shredded columns are expected to store statistics in the same format as a normal Parquet column.
-In general, the engine can only skip a row group or page if all rows in the `variant_value` field are null, since it is possible for a `variant_get` expression to successfully cast a value from the `variant_value` to the target type.
-For example, if `typed_value` is of type `int64`, then the string “123” might be contained in `variant_value`, which would not be reflected in statistics, but could be retained by a filter like `where variant_get(col, “$.field”, “long”) = 123`.
-If `variant_value` is all-null, then the engine can prune pages or row groups based on `typed_value`.
-This specification is not strict about what values may be stored in `variant_value` rather than `typed_value`, so it is not safe to skip rows based on `typed_value` unless the corresponding `variant_value` column is all-null, or the engine has specific knowledge of the behavior of the writer that produced the shredded data.
+Fields of an object can be shredded using a Parquet group for `typed_value` that contains shredded fields.
 
-# Shredding Semantics
+If the value is an object, `typed_value` must be non-null.
+If the value is not an object, `typed_value` must be null.
 
-Reconstruction of Variant value from a shredded representation is not expected to produce a bit-for-bit identical binary to the original unshredded value.
-For example, in a reconstructed Variant value, the order of object field values may be different from the original binary.
-This is allowed since the [Variant Binary Encoding](VariantEncoding.md#object-field-id-order-and-uniqueness) does not require an ordering of the field values, but the field IDs will still be ordered lexicographically according to the corresponding field names.
+Each shredded field in the `typed_value` group is represented as a required group that contains optional `value` and `typed_value` fields.
```

Comment:
Are we making these required just to reduce complexity, it seems like there is potential value in allowing setting the value null at the parent level (maybe not with repetition/definition level, but if we ever revisit this encoding it could be useful).

Reply:
seems like the answer below is yet. it might pay to clarify this is the reason.